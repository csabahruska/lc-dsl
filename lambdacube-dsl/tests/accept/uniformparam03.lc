render modelViewProj = let
  emptyFB           = FrameBuffer (colorImage1 (V4F 0.5 0.0 0.4 1.0))
  rasterCtx         = LineCtx 1.0 LastVertex
  fragmentCtx       = AccumulationContext (ColorOp NoBlending (V4B True True True True))
  vertexShader v    = VertexOut (PrimMulMatVec modelViewProj v) 1.0 () (Flat v)
  vertexStream      = Fetch "stream4" Lines (IV4F "position4")
  primitiveStream   = Transform vertexShader vertexStream
  fragmentStream    = Rasterize rasterCtx primitiveStream
  fragmentShader f  = FragmentOut f
  frame             = Accumulate fragmentCtx PassAll fragmentShader fragmentStream emptyFB
 in ScreenOut frame

main = render (Uni (IM44F "MVP"))
