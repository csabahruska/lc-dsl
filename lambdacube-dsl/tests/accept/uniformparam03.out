Pipeline {backend = OpenGL33, textures = [], samplers = [], targets = [RenderTarget {renderTargets = [(Color,Just (Framebuffer Color))]}], programs = [Program {programUniforms = fromList [("MVP",M44F)], programStreams = fromList [("v",("position4",V4F))], programInTextures = fromList [], programOutput = [("f0",V4F)], vertexShader = "#version 330 core\nuniform mat4 MVP ;\nin vec4 v ;\nflat out vec4 v0 ;\nvoid main() {\nv0 = v;\ngl_Position = ( MVP ) * ( v );\ngl_PointSize = 1.0;\n}\n", geometryShader = Nothing, fragmentShader = "#version 330 core\nflat in vec4 v0 ;\nout vec4 f0 ;\nvoid main() {\nf0 = v0;\n}\n"}], slots = [Slot {slotName = "stream4", slotUniforms = fromList [("MVP",M44F)], slotStreams = fromList [("position4",V4F)], slotPrimitive = Lines, slotPrograms = [0]}], commands = [SetRenderTarget 0,ClearRenderTarget [(Color,VV4F (V4 0.5 0.0 0.4 1.0))],SetRasterContext (LineCtx 1.0 LastVertex),SetAccumulationContext (AccumulationContext {accViewportName = Nothing, accOperations = [ColorOp NoBlending (VV4B (V4 True True True True))]}),SetProgram 0,RenderSlot 0]}